example/src/main/java/io/netty/example/tunechoserver/TunEchoServer.java

```diff
@@ -0,0 +1,448 @@
+        byte[] newTypeAndCode = new byte[2];
+        newTypeAndCode[0] = (byte) newType;
+        newTypeAndCode[1] = (byte) newCode;
+
+        // Update the packet with the new type and code.
+        packetBuf.setBytes(ipHeader.headerLength, newTypeAndCode);
+
+        // Compute a new ICMP checksum based on the updated type and code.
+        int newICMPChecksum = InetUtils.updateInetChecksum(oldTypeAndCodeBytes, 0,
+                newTypeAndCode, 0, 2, icmpHeader.checksum);
+
+        // Update the checksum field in the ICMP header.
+        packetBuf.setShort(ipHeader.headerLength + InetUtils.ICMP_HEADER_OFFSET_CHECKSUM, newICMPChecksum);
+
+        // If the packet is an ICMP packet (vs ICMPv6)...
+        if (ipHeader.ipVersion == InetUtils.IP_VERSION_4) {
+
+            // Read the portion of the packet containing the new ICMP checksum bytes.
+            byte[] newICMPChecksumBytes = new byte[2];
+            packetBuf.getBytes(ipHeader.headerLength + InetUtils.ICMP_HEADER_OFFSET_CHECKSUM, newICMPChecksumBytes);
+
+            // Compute an updated IP checksum.
+            int newIPChecksum = InetUtils.updateInetChecksum(oldICMPChecksumBytes, 0, newICMPChecksumBytes,
+                    0, 2, ipHeader.checksum);
+            newIPChecksum = InetUtils.updateInetChecksum(oldTypeAndCodeBytes, 0, newTypeAndCode, 0, 2, newIPChecksum);
+
+            // Update the checksum field in the IPv4 header.
+            packetBuf.setShort(InetUtils.IPV4_HEADER_OFFSET_CHECKSUM, newIPChecksum);
+        }
+    }
+
+    private static boolean readProps() {
+
+        tunDeviceName = System.getProperty("tun-device-name", "tun-echo-server");
+
+        try {
+            threadPoolSize = Integer.parseInt(System.getProperty("thread-pool-size", "0"));
+            if (threadPoolSize < 0) {
+                throw new IllegalArgumentException();
+            }
+        } catch (Exception ex) {
+            logger.log(Level.SEVERE, "Invalid value specified for thread-pool-size property: " +
+                    System.getProperty("thread-pool-size"));
+            return false;
+        }
+
+        try {
+            readBufSize = Integer.parseInt(System.getProperty("read-buf-size", "2048"));
+            if (readBufSize < 1) {
+                throw new IllegalArgumentException();
+            }
+        } catch (Exception ex) {
+            logger.log(Level.SEVERE, "Invalid value specified for read-buf-size property: " +
+                    System.getProperty("read-buf-size"));
+            return false;
+        }
+
+        if (Boolean.parseBoolean(System.getProperty("use-unpooled", "false"))) {
+            bufAllocator = UnpooledByteBufAllocator.DEFAULT;
+        }
+
+        forceGC = Boolean.parseBoolean(System.getProperty("force-gc", "false"));
+
+        debug = Boolean.parseBoolean(System.getProperty("debug", "true"));
+
+        return true;
+    }
+
+    public static void main(String[] args) throws Exception {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)